File: base/src/main/vsdk/toolkit/environment/geometry/polyhedralBoundedSolid/GeometricStrictFaceIntersectionsStrategy.java
package vsdk.toolkit.environment.geometry.polyhedralBoundedSolid;

public class GeometricStrictFaceIntersectionsStrategy
    implements PolyhedralBoundedSolidValidationStrategy
{
    @Override
    public boolean validate(PolyhedralBoundedSolid solid,
        PolyhedralBoundedSolidNumericPolicy.ToleranceContext numericContext,
        StringBuilder msg)
    {
        return PolyhedralBoundedSolidGeometricValidator
            .validateFaceIntersectionsStrict(solid, numericContext, msg);
    }
}

File: typescript/base/src/main/vsdk/toolkit/environment/geometry/volume/polyhedralBoundedSolid/_GeometricStrictFaceIntersectionsStrategy.ts
//= References:                                                             =
//= [MANT1988] Mantyla Martti. "An Introduction To Solid Modeling",         =
//=     Computer Science Press, 1988.                                       =

import type { StringBuilder } from "../../../../../../java/lang/StringBuilder.js";
import type { PolyhedralBoundedSolid } from "./PolyhedralBoundedSolid.js";
import { PolyhedralBoundedSolidGeometricValidator } from "./PolyhedralBoundedSolidGeometricValidator.js";
import type { ToleranceContext } from "./PolyhedralBoundedSolidNumericPolicy.js";
import type { _PolyhedralBoundedSolidValidationStrategy } from "./_PolyhedralBoundedSolidValidationStrategy.js";

/**
Checks the non-self-intersection requirement imposed on boundary models in
[MANT1988].15.2, criterion 3, using the geometric intersection tools of
chapter [MANT1988].13.
*/
export class _GeometricStrictFaceIntersectionsStrategy implements _PolyhedralBoundedSolidValidationStrategy {
    /**
    Validates that distinct faces only meet in the ways allowed by
    [MANT1988].15.2, criterion 3, relying on the geometric tests discussed in
    chapter [MANT1988].13.
    */
    public validate(solid: PolyhedralBoundedSolid, numericContext: ToleranceContext, msg: StringBuilder): boolean {
        return PolyhedralBoundedSolidGeometricValidator.validateFaceIntersectionsStrict(solid, numericContext, msg);
    }
}
